framework/src/times/FunctorTimes.C

```diff
@@ -0,0 +1,57 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FunctorTimes.h"
+
+registerMooseObject("MooseApp", FunctorTimes);
+
+InputParameters
+FunctorTimes::validParams()
+{
+  InputParameters params = Times::validParams();
+  params += NonADFunctorInterface::validParams();
+  params.addClassDescription(
+      "Times created by evaluating a functor at the (0,0,0) point and the current time");
+
+  params.addRequiredParam<MooseFunctorName>("functor", "Functor to evaluate to provide the time");
+  params.addParam<MooseFunctorName>("factor", 1, "Factor to multiply the evaluated time with");
+
+  // Times are known for all processes already
+  params.set<bool>("auto_broadcast") = false;
+  // Timestep_begin seems like a decent default here
+  params.set<ExecFlagEnum>("execute_on") = EXEC_TIMESTEP_BEGIN;
+
+  return params;
+}
+
+FunctorTimes::FunctorTimes(const InputParameters & parameters)
+  : Times(parameters),
+    NonADFunctorInterface(this),
+    _functor(getFunctor<Real>("functor")),
+    _factor(getFunctor<Real>("factor"))
+{
+}
+
+void
+FunctorTimes::initialize()
+{
+  // Locate the origin on the mesh
+  Point p(0, 0, 0);
+  auto pl = _fe_problem.mesh().getMesh().sub_point_locator();
+  auto * elem = (*pl)(p);
+  if (!elem)
+    mooseError("Origin point not in local mesh, cannot evaluate the functor there");
```

Comment:
What if you just chose the ID 0 element? @roystgnr what happens with point locators on distributed meshes? My suggestion will only be globally consistent on replicated meshes. Wondering whether this current implementation has the same issue

Reply:
If the point is on a local or ghosted element, the point locator will return one of those.  If it's only on remote elements, the point locator will return null.

An element with id 0 might not exist either; element deletion and `allow_renumbering(false)` are things.

If you need an arbitrary element on every processor that'll be computing on any elements then `*(mesh.elements_begin())` will get you one (though you'll still need to check that `elements_begin() != elements_end()` to handle distributed meshes where a processor has no elements).

If you need an arbitrary element on every processor and it's the same element on every processor ... this can't be done in general.

Reply:
Would it make sense to only evaluate on processor(s) which contain `Point(0)`, then find the lowest-ranked one of those and `broadcast()` from there?

Reply:
Second solution makes sense. 
However the real solution here is a MeshArg argument to functor. 
For all the functors for which it would be defined (functions? with an extra parameter maybe to specify the point)

Do we really need this now though

Reply:
> Do we really need this now though

I don't know to what you're referring to, but I'm fine leaving this as is as long as you add an error check or at least a warning check in this class for whether the mesh is distributed

Reply:
added an errorIfDistributed